x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googlevertexai/completion/GoogleVertexAiChatCompletionModel.java

```diff
@@ -65,8 +66,37 @@ public GoogleVertexAiChatCompletionModel(
             serviceSettings
         );
         try {
-            this.streamingURI = buildUriStreaming(serviceSettings.location(), serviceSettings.projectId(), serviceSettings.modelId());
-            this.nonStreamingUri = buildUriNonStreaming(serviceSettings.location(), serviceSettings.projectId(), serviceSettings.modelId());
+            var provider = serviceSettings.provider();
+            var uri = serviceSettings.uri();
+            var streamingUri = serviceSettings.streamingUri();
+            // If provider is set, uri or streamingUri must be set. If provider is not set, location, projectId and modelId must be set
+            if (provider != null && (uri != null || streamingUri != null)) {
+                // If both URIs are provided, use them as is. If only one is provided, use it for both streaming and non-streaming
+                this.nonStreamingUri = Objects.requireNonNullElse(uri, streamingUri);
+                this.streamingURI = Objects.requireNonNullElse(streamingUri, uri);
```

Comment:
Can you remind me, there are legitimate cases where a user would only have a single url right? Could you add a few examples in the comments of when we'd expect to have both urls and when we'd expect to only have one?